api/app/controllers/api/v1/applications_controller.rb

```diff
@@ -36,7 +43,7 @@ def create
         # PUT/PATCH /applications/:id
         def update
             application = Application.find(params[:id])
-            matching = ApplicantDataForMatching.find(application[:id])
+            matching = ApplicantDataForMatching.find_by!(application_id: application[:id])
```

Comment:
Again, `Application` has a `has one` relationship to `ApplicantDataForMatching`. Couldn't we just do:
`application.ApplicantDataForMatching`?

Reply:
Good point. Thanks for catching that. I'll change it.

Reply:
Done.